src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -0,0 +1,164 @@
+from __future__ import annotations
+
+import abc
+from collections.abc import Callable
+
+from databricks.sdk.service.workspace import ObjectType, ObjectInfo, ExportFormat
+from databricks.sdk import WorkspaceClient
+
+
+class Dependency:
+
+    @staticmethod
+    def from_object_info(object_info: ObjectInfo):
+        assert object_info.path is not None
+        return Dependency(object_info.object_type, object_info.path)
+
+    def __init__(self, object_type: ObjectType | None, path: str):
```

Comment:
```suggestion
    def __init__(self, object_type: ObjectType, path: str):
```
let's simplify the handling and ignore objects with no types